### ISSUE TITLE
path method ignored on Image and Avatar

### ISSUE THREAD

**Issue by Woeler** [laravel/nova-issues#1110]
- Laravel Version: 5.7.*
- Nova Version: 1.2.0
- PHP Version: 7.2

### Description:
Hello all,

I'm trying to add an image to my Nova resource, but for some reason it is ignoring the path method.

I have a resource, for which the images are stored in the public storage (yes, the storage has been linked) in the folder `skills`. The name of the image file is stored in the `icon` database row of the `Skill` model class.

This is my code:
```
public function fields(Request $request)
    {
        return [
            ID::make()->sortable()->hideFromIndex(),
            Text::make('Name')->sortable()->rules('required'),
            Avatar::make('Icon')->disk('public')->path('skills'),
        ];
    }
```

However, all image url's are linking to the storage root of public, and I cannot seem to figure out why.

The odd thing is, that on another Nova resource, it is working totally fine. Here is the code for the working resource:
```
public function fields(Request $request)
    {
        return [
            ID::make()->sortable(),
            Text::make('name')->sortable(),
            Text::make('Zone'),
            BelongsTo::make('Set', 'set'),
            Image::make('Image')->disk('public')->path('craftinglocations'),
        ];
    }
```

I tried using `Image` and `Avatar` interchangably, but for some reason unknown to me the top class fails, but the bottom class succeeds. Any ideas?

**Comment by dkulyk** [comment]
What is stored in the DB for both fields?

**Comment by Woeler** [comment]
@dkulyk Hey, both are stored exactly the same, just the filename.

**Comment by dkulyk** [comment]
You need to link root of the public disk. Path is additional path related to the root.

**Comment by dkulyk** [comment]
In DB must be filename with path

**Comment by Woeler** [comment]
No, that is not true. Because as I said, I have a resource that is working. See here:
![image](https://user-images.githubusercontent.com/18422096/49676868-02859e00-fa7c-11e8-83a6-281dd2b17866.png)

I also do not want to store a path in the database, because that would mean if I ever were to change some storage logic, I would have to update all my model tables.

**Comment by dkulyk** [comment]
May you show disk options and ls -l for link?

**Comment by Woeler** [comment]
![image](https://user-images.githubusercontent.com/18422096/49677020-ca328f80-fa7c-11e8-809f-e03c0408ffc0.png)

**Comment by dkulyk** [comment]
Files you upload from nova or you own code?

**Comment by Woeler** [comment]
I placed them there manually, but they have exactly the same permissions and owners as the files uploaded.

**Comment by dkulyk** [comment]
Nova file use path for store file only. It store path with filename in DB. On view file field get filepath from DB and get url for it from the Storage

**Comment by dkulyk** [comment]
You can try to set resolveCallback for adding the path
